### ISSUE TITLE
Expand shield coverage beyond the United States

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#334]
The turn banner and step table show graphical route shields for a number of route networks in the United States. Shields.plist maps `ref` prefixes to file name patterns on Wikimedia Commons. Commons has a much larger repertoire of shields, but we’ve run up against the limits of parsing the `ref` tag on ways, and even what we have now is prone to false positives. For example:

* [`CA`](https://github.com/mapbox/mapbox-navigation-ios/blob/1677c6479a2a293172a4a4d1e3ea45e5c1e3707b/MapboxNavigation/Resources/Shields.plist#L29-L30) may denote California or Cantabría, Spain
* [`NH`](https://github.com/mapbox/mapbox-navigation-ios/blob/1677c6479a2a293172a4a4d1e3ea45e5c1e3707b/MapboxNavigation/Resources/Shields.plist#L83-L84) may denote New Hampshire or National Highway (e.g., in India)
* [`CO`](https://github.com/mapbox/mapbox-navigation-ios/blob/1677c6479a2a293172a4a4d1e3ea45e5c1e3707b/MapboxNavigation/Resources/Shields.plist#L31-L32) may denote Colorado or a county road.
* We can’t add support for German shields to the turn banner, because we can’t tell whether to use a German, Austrian, Australian, or Chinese shield for [`A`](https://en.wikipedia.org/wiki/A_road).

Unfortunately, there isn’t a better way to distinguish shields internationally until either Project-OSRM/osrm-backend#482 or Project-OSRM/osrm-backend#333 is implemented. This issue affects routes in the U.S. as well: state trunkline highways in Michigan have the prefix `M`, which is identical to motorways in several other countries.

For more background on graphical route shields and why they matter in navigation, see gravitystorm/openstreetmap-carto#508 and [this OSM diary entry](http://www.openstreetmap.org/user/Minh%20Nguyen/diary/39123), or watch [this talk from State of the Map 2011](https://wiki.openstreetmap.org/wiki/SotM_2011_session:_Shields_Up!).

/ref https://github.com/mapbox/mapbox-navigation-ios/issues/286#issuecomment-309649781
/cc @ericrwolfe @danpat @freenerd

**Comment by 1ec5** [comment]
To illustrate the scale of worldwide shield coverage, MapKit on macOS 10.12 ships with an _initial_ set of 27 _packs_ of shields (/System/Library/PrivateFrameworks/GeoServices.framework/Versions/A/Resources/InitialResources/), each pack containing the shield assets needed for a particular region of the world. 0-0-1-north-america-shields-index-23.shieldindex contains 70 shield designs, while 3-2-2-oceania-shields-index-7.shieldindex contains 17. In all, MapKit comes with 9,526 shield designs out of the box, multiplied by screen resolution (1×, 2×, 3×) and image size (XS, S, M, L, XL). As you browse additional regions of the world, MapKit presumably downloads additional shield packs automatically.

**Comment by 1ec5** [comment]
For a more realistic (and relevant) figure, I took a look at Wikimedia Commons, which is our current source of shield images. Commons hosts [305 template SVG files](https://petscan.wmflabs.org/?language=commons&project=wikimedia&depth=25&categories=Highway%20shield%20templates&ns%5B6%5D=1&interface_language=en&active_tab=), one per design/reflen combination, from which contributors have generated and uploaded [141,592 SVG images](https://petscan.wmflabs.org/?language=commons&project=wikimedia&depth=25&categories=Diagrams%20of%20route%20signs&ns%5B6%5D=1&interface_language=en&) that have numerals baked into them, one image per route. There are also [229 shield blanks](https://petscan.wmflabs.org/?language=commons&project=wikimedia&depth=25&categories=Highway%20shield%20blanks&ns%5B6%5D=1&interface_language=en&active_tab=), one per design/reflen combination, each of which is a vector or raster images that can be composited with numerals. Note that many of these designs are for the county level and below, which are most likely out of scope for this issue.

**Comment by brsbl** [comment]
@1ec5 looks like both of these have been closed -- still blocked upstream?
>Unfortunately, there isn’t a better way to distinguish shields internationally until either Project-OSRM/osrm-backend#482 or Project-OSRM/osrm-backend#333 is implemented.

**Comment by 1ec5** [comment]
Now blocked by:

* Project-OSRM/osrm-backend#3329
* Project-OSRM/osrm-backend#3304
* Project-OSRM/osrm-backend#4434

and the selection or generation of shield artwork.

**Comment by 1ec5** [comment]
Still blocked upstream.
